add documentation

brainrape opened the pull request.

paf31 commented:
Looks great. Does this include any verbatim content from the reference?

brainrape commented:
Only a few parts of sentences match. I believe it is within fair use.

paf31 commented:
Ok, I just want to make sure we give attribution correctly. Maybe we should mention CC-SA, and then in the LICENSE file, mention that the _documentation_ is licensed the same way.

brainrape commented:
Updated. To keep it simple, now it can be reasonably said that it doesn't include any verbatim content from the reference. Also added instance docs in `Char`.

paf31 commented:
:1: Thanks!
